### ISSUE TITLE
ibm5170 driver cannot read hard disk

### ISSUE THREAD

**Issue by tinue** [mamedev/mame#2976]
I have a harddisk image that works fine with the driver 'atvga'. The same image has issues with the driver 'ibm5170'. The main difference between 'atvga' and 'ibm5170' seems to be the BIOS used.
On the surface it looks like the 'IDE Fixed Drive Adapter' does not quite work with the original IBM AT BIOS, but it works fine with the AMI BIOS that is used in 'atvga'.

Symptom: The disk can be written fine (FDISK, FORMAT, COPY etc.), but reading mostly (but not always) fails. For example, a ``CHKDSK C:`` yields:
```
Volume DOS     created Dec 27, 2017 12:51p
Disk error reading FAT 1
Disk error reading FAT 2
Not ready error reading drive C.
Abort, Retry, Ignore?
```

This on the other hand works, even though FAT 1 and 2 are supposedly unreadable:
```
A>echo foo > c:bar.txt

A>type c:bar.txt
foo
```

Additional background: The image was created with ``chdman createhd -o AT30MB.hd -chs 733,5,17``
The 'ibm5170' image was initially booted with the diag207 floppy image. The hard disk type was then set to type '30'. Then, the system was booted with a DOS 3.3 floppy image in drive A.
I also tried a type 2 geometry, with identical results.

**Comment by PeladoFeo** [comment]
Maybe the original IBM AT only works with MFM hard disks...

**Comment by angelosa** [comment]
> Additional background: The image was created with chdman createhd -o AT30MB.hd -chs 733,5,17 The 'ibm5170' image was initially booted with the diag207 floppy image. The hard disk type was then set to type '30'.

I'm not sure what _atvga_ maps to nowadays, but regardless I don't think your geometry is right with the cylinder part (it's supposedly 918?), 
![0000](https://user-images.githubusercontent.com/9251271/107159257-2c55ea80-698f-11eb-95fc-16a1dd62dc95.png)

**Comment by tinue** [comment]
Yes, there is a mixup somewhere. I don't know whether I actually configured the disk wrongly, or if I just mistyped in my posting. 733/5/17 would be a type 8.
I most likely mistyped: "30" is the disk capacity (30MB). Also I did try with a type 2 (615/4/17), and got the same results.

**Comment by tinue** [comment]
Apparently the emulated hard disk is "too fast", and triggers a race condition in the AT BIOS. This case is nicely written up [here](https://ia801506.us.archive.org/14/items/ibm-5170-type-3-bios-cover/IBM%205170%20Type%203%20BIOS%20Patch%20Summary.txt).

It would appear that the hard disk emulation should slow down the disk a bit, and then the original AT BIOS would work.
